syntax = "proto3";

package payment_repository;

service PaymentRepositoryService {
  // Stores invoice details.
  rpc StoreInvoice (StoreInvoiceRequest) returns (StoreInvoiceResponse) {}

  // Retrieves a specific invoice.
  rpc RetrieveInvoice (RetrieveInvoiceRequest) returns (InvoiceData) {}

  // Retrieves all invoices for a specific user.
  rpc GetUserInvoices (GetUserInvoicesRequest) returns (UserInvoicesResponse) {}
}

message StoreInvoiceRequest {
  InvoiceData invoice = 1;
}

message StoreInvoiceResponse {
  bool success = 1;
  int64 invoiceId = 2;
}

message RetrieveInvoiceRequest {
  int64 invoiceId = 1;
}

message InvoiceData {
  int64 invoiceID = 1;
  double price = 2;
  int64 orderID = 3;
  int64 userId = 4;
  string fiscalAddress = 5;
  CardDetails cardDetails = 6;
}

message GetUserInvoicesRequest {
  string userId = 1;
}

message UserInvoicesResponse {
  repeated InvoiceData invoices = 1;
}
